packages/application-shell/src/components/fetch-project/fetch-project.js

```diff
@@ -77,6 +77,7 @@ class FetchProject extends React.Component {
                 permissions: mapAllAppliedToObjectShape(
                   data.project.allAppliedPermissions
                 ),
+                dataFences: data.project.allAppliedDataFences,
```

Comment:
the `mapAll*` maps permissions and action rights to 

```js
{ [permission|actionRight]: true }
```

which is not my desired shape for dataFences. However it makes a whole lot of sense to group each dataFence, which is WIP now

Reply:
1. We will "leak" the `__typename` here into the React context if I am not mistaken. 
2. The mapping from an array to object shape for the other two has benefits
   - No `includes` for matching just object access
   - Easier to read in React DevTools

Reply:
> We will "leak" the __typename here into the React context if I am not mistaken.

yeah that this is true, but would this be a pitfall even if we strip it off before we set it on the `applicationContext`?

---

> No includes for matching just object access

we have to do a find. for the dataFence, it will be hard to read either way. because we're introducing another "dimension" to match with `storekey + permission`. so it won't be smooth to read it even if we flatten it to an `{}`..